### ISSUE TITLE
Add warning for private_class_method used without arguments

### ISSUE THREAD

**Issue by agrimm** [rubocop-hq/rubocop#3299]
RuboCop does not explain that using `private_class_method` without any arguments is ineffective.

Failing this, the error given by `Lint/IneffectiveAccessModifier` should be less ambiguous, explicitly stating that you have to supply the method name to `private_class_method`.

---
## Expected behavior

RuboCop tells users that using `private_class_method` without any arguments is ineffective.
## Actual behavior

RuboCop either gives no warning, or gives a confusing message about `private`.
## Steps to reproduce the problem

Given the following code:

```
# Example for RuboCop
class A
  private_class_method

  def self.c
    puts 'Class method which the author meant to be private but is not private'
  end
end
```

RuboCop will not produce any warnings.

Given the following code:

```
# Example for RuboCop
class A
  private

  def self.c
    puts 'Class method which the author meant to be private but is not private'
  end
end
```

RuboCop gives the confusing message, which may cause people to think having a `private_class_method` without any arguments will work. (See http://stackoverflow.com/a/4953043/38765 for an example of someone assuming that approach should work, though this example predated RuboCop giving this warning)

```
$ rubocop -D private_example.rb 
Inspecting 1 file
W

Offenses:

private_example.rb:3:3: W: Lint/UselessAccessModifier: Useless private access modifier.
  private
  ^^^^^^^
private_example.rb:5:3: W: Lint/IneffectiveAccessModifier: private (on line 3) does not make singleton methods private. Use private_class_method or private inside a class << self block instead.
  def self.c
  ^^^

1 file inspected, 2 offenses detected
```
## RuboCop version

```
$ rubocop -V
0.41.2 (using Parser 2.3.1.2, running on ruby 2.4.0 x86_64-darwin14)
```
